component/prometheus/relabel/relabel_test.go

```diff
@@ -120,29 +79,30 @@ func TestNil(t *testing.T) {
 	require.NotNil(t, relabeller)
 	require.NoError(t, err)
 
-	fm := prometheus.NewFlowMetric(0, labels.FromStrings("__address__", "localhost"), 10)
-	relabeller.Receive(time.Now().Unix(), []*prometheus.FlowMetric{fm})
-
-	newFm := prometheus.NewFlowMetric(0, labels.FromStrings("__address__", "localhost"), 20)
-	relabeller.Receive(time.Now().Unix(), []*prometheus.FlowMetric{newFm})
-	require.True(t, fm.GlobalRefID() == newFm.GlobalRefID())
+	lbls := labels.FromStrings("__address__", "localhost")
+	relabeller.relabel(0, lbls)
 }
 
 func BenchmarkCache(b *testing.B) {
-	rec := &prometheus.Receiver{
-		Receive: func(timestamp int64, metrics []*prometheus.FlowMetric) {
-		},
-	}
 	l := log.NewSyncLogger(log.NewLogfmtLogger(os.Stderr))
 
-	relabeller, _ := New(component.Options{
+	fanout := prometheus.NewFanout(func(ref storage.SeriesRef, l labels.Labels, tt int64, v float64) (storage.SeriesRef, labels.Labels, int64, float64, error) {
+		if !l.Has("new_label") {
+			panic("must have new label")
+		}
```

Comment:
Remove the panic please (also I'm not sure what new_label means)

Reply:
Oh, this is in test code, nvm :) 

This would be more idiomatic for our existing tests: 

```go
requires.True(t, l.Has("new_label"), "new_label must be provided") 
```

Reply:
Can benchmarks access the testing.T?

Reply:
oh, it'd be `b` here then. `require` can be given any `testing.T` or `testing.B` instance

Reply:
Doh!